eth/tracers/tracer.go

```diff
@@ -541,6 +541,13 @@ func (jst *Tracer) CaptureStart(from common.Address, to common.Address, create b
 	return nil
 }
 
+func (jst *Tracer) CaptureExtraContext(inputs map[string]interface{}) error {
```

Comment:
@iquidus @meowsbits I am thinking to make the `map[string]interface{}` a Struct and parse it in here with reflect. Do you think it's ok for now like it is? Is it overkill to use reflect for this?

Reply:
For now I think it would be overkill... although worth considering for a refactor later on. The `Tracer.ctx` field is used in several ways that seem to depend on the type-agnosticism of the `map[string]interface{}`, eg. `txTraceTask.extraContext`.

Reply:
That's true, originally `Tracer.ctx` holds transaction context. I have added `CaptureExtraContext` in order to be able to extend context with block related stuff. For this reason I was thinking to make only the `inputs` in `CaptureExtraContext` a struct. Though at the end, they will be added down to the `Tracer.ctx` being a `map[string]interface{}`